spec/support/page_helpers.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module PageHelpers
+  def reload_page_until_timeout!(text:)
+    Timeout.timeout(Settings.timeouts.workflow) do
+      loop do
+        page.evaluate_script('location.reload(true);')
+        break if page.has_text?(text)
+      end
+    end
+  end
```

Comment:
I think you should maybe have `reload_page_until_text_or_timeout` and `reload_page_until_link_or_timeout` ...